### ISSUE TITLE
Decouple Controllers, Don't or Can't?  

### ISSUE THREAD

**Issue by EresDev** [mrkrstphr/cleanphp-example#3]
Great book and I am learning a lot from it. However,  clean architecture targets maximum decoupling. That is why we should decouple controllers as well. Frameworks like Symfony allow you to write controller class without implementing any interface or extending a class. However, to use helper methods, you need to extend. We can write adapter for helpers that we need. But I am not sure how it will play out with other frameworks. 

I will appreciate your thoughts on it.

**Comment by mohamed-abdul-fattah** [comment]
IMHO, controllers are embedded within the application layer, which is tightly coupled with the framework and should not be decoupled.
Controllers by design uses the framework request and response classes, plugins and utilize helpers to deliver data to the presentation layer, and decoupling controllers would require us to wrap all these dependencies which seems like implementing a framework inside a framework.
To sum up, frameworks provide the application layer which plays the composer role that composes our domain and services layers with the presentation and infrastructure layers.
